### PR TITLE
Support RGB(A) array as colormap input type

### LINKED ISSUE #3141: Feature: accept RGB(A) tuple as colormap
## 🚀 Feature
Just as our `add/view_image()` function accepts a string "red" for the `colormap` value (meaning you want a colormap that goes from black to red), it should accept an RGBA tuple, meaning you want a linear colormap that goes from black to whatever color that is.

The alternative is to import either napari or vispy to create a colormap object manually (which means heavy dependencies for plugins is that's the only thin they're using it for)

originally mentioned in https://github.com/napari/napari/pull/2936#issue-679256433

### PR DESCRIPTION
**Pull request by andy-sweet**
# Description
This changes the implementation of `ensure_colormap` so that it converts RGB(A) array-likes (i.e. `tuple`, `list`, `np.ndarray`) as user input that can be coerced into a napari `Colormap` object. If one color value is provided, this creates a colormap that interpolates between black and the specified single color. If multiple color values are provided, this creates a colormap from those. In both cases, the colormap is unnamed (i.e. `Colormap.name` has the default `custom`) and so the `AVAILABLE_COLORMAPS` dictionary is not updated.

The logic of `ensure_colormap` is already complex, and this adds more behavior and complexity to it, but I didn't see obvious ways to simplify things.

In #2936, I mentioned that sometimes it could be useful to raise instead of warn when the input colormap type is unsupported (e.g. for some Pydantic model validators). I thought about making a change here to make that configurable (e.g. by adding a boolean parameter to condition warning vs. erroring), but I think a better approach is to add that configuration when actually needed.

I did however update and change some documentation appropriately.

I also moved the definition of `ColorType` to avoid some import dependency cycles - it's new location seems appropriate, but let me know if you have better ideas or locations for that. I also redefined `ColorType` to use `str` instead of `AnyStr`, as `bytes` strings are not currently supported by `transform_color`.

## Type of change
- [ ] Bug-fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References
Closes #3141 

# How has this been tested?

- [ ] I added new tests in `test_colormaps` to check that the single color and multi color array likes are supported.
- [ ] example: all existing related tests continue to pass.

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

**Review by sofroniewn — APPROVED** [review]
LGTM! Thanks!!

**Comment by sofroniewn** [comment]
@andy-sweet do you know why the CI isn't running for this PR? Can you maybe make another commit? Thanks!

**Comment by andy-sweet** [comment]
> @andy-sweet do you know why the CI isn't running for this PR? Can you maybe make another commit? Thanks!

Yep, I used `[no ci]` in my last commit, which was just a change to docstrings/typing. I just merged `main` into this and pushed to trigger all the checks.

**Comment by sofroniewn** [comment]
> Yep, I used [no ci] in my last commit, which was just a change to docstrings/typing. I just merged main into this and pushed to trigger all the checks.

Ah, i got it know

@tlambert03 any thoughts on this before merge?

File: napari/utils/colormaps/colormap_utils.py [hunk 1]
```diff
@@ -1,18 +1,25 @@
...
+from vispy.color import Color, ColorArray, get_colormap, get_colormaps
 
 from ..translations import trans
 from .bop_colors import bopd
 from .colormap import Colormap
+from .standardize_color import transform_color
 from .vendored import cm, colorconv
 
+# All parsable input color types that a user can provide
+ColorType = Union[List, Tuple, np.ndarray, str, Color, ColorArray]
```
Comment (tlambert03) [inline napari/utils/colormaps/colormap_utils.py:17]: still hoping for the day that we move to `pydantic.types.Color` :)

**Review by tlambert03 — APPROVED** [review]

File: napari/utils/colormaps/colormap_utils.py (hunk 1 above)
Comment (sofroniewn) [reply napari/utils/colormaps/colormap_utils.py:17]: yes for sure - maybe @andy-sweet can make an issue for that if we don't have one already. i think can come in followup PRs though :-)

**Pull request merged**